tests/ut/language/parser/test_error_cases.py

```
@@ -395,10 +395,36 @@ def bad_pl_while(n: pl.Scalar[pl.INT64]) -> pl.Scalar[pl.INT64]:
-                    y = pl.yield_(x_iter + 1)  # noqa: F841
+                    y = pl.yield_(x_iter + 1)
                 return y  # noqa: F821
 
 
+class TestYieldLHSRequired:
+    """`pl.range(init_values=...)` and `pl.while_(init_values=...)` both
+    require an assignment-form `pl.yield_(...)` so the LHS supplies the
+    post-loop binding name (uniform convention across both loop forms)."""
+
+    def test_pl_range_bare_yield_with_init_values_rejected(self):
+        with pytest.raises(ParserSyntaxError, match="requires an assignment-form pl.yield_"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)